### DISCUSSION TITLE
How to use LDAP connector?

### DISCUSSION THREAD

**Discussion by Harold7665** [atoti/atoti#513] [Q&A] [answered]
Hello,
I would like to use the LDAP function and as with our Java API, I was expecting to give it the same information, including a password. However I can not see how to do it at the moment, could you give me more information?
Thanks.

**Comment by tibdex** [comment, marked as answer]
Have your read [our docs](https://docs.atoti.io/0.6.4/lib/atoti.config.authentication.ldap.html#atoti.config.authentication.ldap.LdapConfig) on this?

> I was expecting to give it the same information, including a password.

Which password do you expect to have to pass to the atoti session?

Securing an atoti session with LDAP consists in delegating authentication to an existing LDAP server: it's when users log in that they have to enter their username and password. These credentials are not needed when starting the atoti session.

**Reply by mwebborn-av** [reply]
Hi, 
Following up on this, we need to specify a user and password to access ActiveDirectory at all before a UserSearch is executed for an individual user. 
Spring LDAP allows setting a username and password to the LDAP connection in its ContextSource. See this class which the default LDAP context source extends: https://docs.spring.io/spring-ldap/docs/current/apidocs/org/springframework/ldap/core/support/AbstractContextSource.html

Typically in Java we configure the context source like so: 
```    
   @Bean
    public ContextSource contextSource() {
        final LdapContextSource lcs = new LdapContextSource();
        lcs.setUrl(ldapUrl);
        lcs.setUserDn(ldapUserDn);
        lcs.setPassword(ldapPassword);
        lcs.setBase(ldapBase);
        return lcs;
    }
```

Is there any way currently to do that? Feels like we might be missing it in the current atoti LdapConfig.

Thanks

**Reply by tibdex** [reply]
Hi, I've created https://github.com/atoti/atoti/issues/664. In the meantime, I never tried it but, could you try adding the credentials in `LdapConfig.url`? Something like that: `LdapConfig(url=f"ldap://{username}:{password}@therestoftheurl")`

**Reply by mwebborn-av** [reply]
Hi, thanks for the suggestion. I've tried with that now but the URL throws a NamingException in the constructor of com.sun.jndi.ldap.LdapUrl. I guess it doesn't allow that format.

**Reply by mwebborn-av** [reply]
> Hi, I've created #664. In the meantime, I never tried it but, could you try adding the credentials in `LdapConfig.url`? Something like that: `LdapConfig(url=f"ldap://{username}:{password}@therestoftheurl")`

Any ideas when that issue might get picked up? This is a blocker for getting LDAP set up for a client at the moment.

**Reply by tibdex** [reply]
We'll try to start working on it next week.

**Reply by mwebborn-av** [reply]
We've deployed this fix on site and I can confirm that it's working as expected now. Thanks for the fix!

**Comment by Harold7665** [comment]
Hello,
Thank you for your answer, it seems that the customer had another issue related to his own access to LDAP.
